standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/common/ndv/hll/HLLSparseRegister.java

```diff
@@ -148,8 +148,12 @@ public int encodeHash(long hashcode) {
     }
   }
 
-  public int getSize() {
-    return sparseMap.size() + tempListIdx;
+  public boolean isSizeGreaterThan(int s) {
+    if (sparseMap.size() + tempListIdx > s) {
+      mergeTempListToSparseMap();
```

Comment:
The tempList array was added for insertion performance (at the cost of slight misestimation when there are duplicates in the temp list) to quickly buffer up 1024 elements. This is negating that by merging it to sparse map more frequently. We might as well add it to sparse map directly instead of tempList so that getSize() becomes a constant time operation. The sparse register will switch to dense around after 3200 elements so sparsemap will not be a hot loop. I would recommend removing tempList from sparse register that way there will be less branching.

Reply:
we are using some "sizeOptimized" version; which switches to dense around ~130 elements

I'll reply back to removal/etc on the jira

Reply:
~130 elements? Did encoding switch threshold change? 
Can you please push commit to github PR for easier review?

Reply:
we are using:
* sizeOptimized => p=10
* bitpacking is enabled by default 
formula to count the threshold in this case is: 
```
2**p * 6/8/5 = ~150
```
https://github.com/apache/hive/blob/d91cc0cd84b7d0ecc0f29d44b109b46e21194eec/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/common/ndv/hll/HyperLogLog.java#L116

I also found that a little too few...but that's what it is...

all changes are here; this conversation is on a diff which is "outdated"

Reply:
ah oh.. i did the math with p=14 and forgot that we switched to 10 in hive.

Reply:
also can we remove fastutil dependency? For such small encoding switch threshold I don't think fastutils fast hashmap implementation is worth it.